crates/sui-json-rpc/src/api/mod.rs

```diff
@@ -39,16 +41,20 @@ mod read;
 mod transaction_builder;
 mod write;
 
-pub const QUERY_MAX_RESULT_LIMIT: usize = 50;
+const RPC_QUERY_MAX_RESULT_LIMIT: &str = "RPC_QUERY_MAX_RESULT_LIMIT";
+const DEFAULT_RPC_QUERY_MAX_RESULT_LIMIT: usize = 50;
+
+pub static QUERY_MAX_RESULT_LIMIT: Lazy<usize> = Lazy::new(|| {
+    read_size_from_env(RPC_QUERY_MAX_RESULT_LIMIT).unwrap_or(DEFAULT_RPC_QUERY_MAX_RESULT_LIMIT)
+});
+
```

Comment:
This is great, thank you.
One small point: I believe this is better placed [NodeConfig](https://github.com/MystenLabs/sui/blob/main/crates/sui-config/src/node.rs#L42) so it can be specified in the yaml file like all other configs.
Much easier for operators to colocate all configs.

Reply:
can you elaborate why it's better to be put in a NodeConfig? We have quite a lot configs in env variables already.

Reply:
I didn't realize other configs were already env vars. Yeah I guess we can let this slide now but we should really collocate them as it makes it easier to know what all the configs are available by looking at one place e.g NodeConfig.